### ISSUE TITLE
Suggestion: Make the run button a toggle?

### ISSUE THREAD

**Issue by therealHelDM** [anegostudios/VintageStory-Issues#59]
so uhm, yeah, Sept suggested I put this here. 

When you run, you need to constantly hold the run button. It's also the annoying 'ctrl' button by default.
Why not make it a run toggle? One that switches off when you let go of the W key? 
(so it's not really a toggle, but more of an instigator?) does this make sense?

**Comment by Heptagon-ru** [comment]
An auto-run button would be also nice.

**Comment by copygirl** [comment]
Should be one option of many, instead of a new default (without alternatives).

One control scheme could have running bound to touble-tapping W.

(In my opinion, running should only possible while going forward anyway. "Sprinting" sideways or backwards could perhaps perform a dash, instead. Only really useful if combat becomes more interesting?)

**Comment by Heptagon-ru** [comment]
Dashing wolves? Sweet!

**Comment by therealHelDM** [comment]
I partially agree copy. Double tapping W or toggling the run with a single button push while only going forward is sensible.

I'm hard pressed to find good reasons to not make it a default setting. Why would someone prefer holding the run key in stead of pressing it once?

@hept/sept whatever you're callled nowadays ;-): an autorun function like in most mmo's doesn't really make sense. Especially with the envisioned stamina system.. (imho)

**Comment by Heptagon-ru** [comment]
Double tapping W may be bad for caving, when you walking an edge above a deep pit and don't need sprint to suddenly enable. Personally I hate double W instead of Shift.
Toggling sprint may be bad for fights, when you need to accurately control your movements - run when increasing the distance and walk when hitting to get accurate blow.
Autorun is very useful for a server with many long roads, like the official one. We have a road to -3000 to one base, a skybridge ... Also when prospecting ore and shafting you need to climb down and up a lot.

That's why all the sprint modifications are useful and should be switchable in options, and why autorun is useful :-P

**Comment by copygirl** [comment]
> I'm hard pressed to find good reasons to not make it a default setting. Why would someone prefer holding the run key in stead of pressing it once?

I'm not saying it shouldn't be the default. But people who do prefer it (Hi `o/`) might want to switch back to the current behavior. I feel like I'm causing the running / sneaking by the act of keeping the button pressed down, so I never liked toggle controls much.

> Double tapping W may be bad for caving, when you walking an edge above a deep pit and don't need sprint to suddenly enable. Personally I hate double W instead of Shift.

Again, that's why it should be an **option** for people who might prefer this, being used to this in other games. Sprinting could easily be disabled by sneaking so you can't accidentally run off a block, which is something that sneaking is used for anyway.

> Autorun is very useful for a server with many long roads, like the official one. We have a road to -3000 to one base, a skybridge ... Also when prospecting ore and shafting you need to climb down and up a lot.

Unrelated to the issue, but I think in cases like this it would be more interesting for gameplay to offer alternative transporation methods such as by horse, carriage, train, or for example building an mechanical powered elevator, rather than empowering the player themself. At least in the long run.

**Comment by therealHelDM** [comment]
Alright, fair point. I didn't look at it like that. 
So the suggestion is going to be altered to:
An extra option in the settings to switch running from continuously needing to press a button, to a 'soft toggle' button.

I agree with the alternate modes of transportation :) gimme horses, boars, wolves even :)

**Comment by sirStroam** [comment]
No longer supporting suggestions here. Please move to forum suggestions.
https://www.vintagestory.at/forums/forum/5-suggestions/

**Comment by sirStroam** [comment]
duplicate of this
https://github.com/anegostudios/VintageStory-Issues/issues/223
